apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Span.java

```diff
@@ -219,4 +232,24 @@ public Span withStacktrace(Throwable stacktrace) {
         this.stacktrace = stacktrace;
         return this;
     }
+
+    @Override
+    public void incrementReferences() {
+        if (transaction != null) {
+            transaction.incrementReferences();
+        }
+        super.incrementReferences();
+    }
+
+    @Override
+    public void decrementReferences() {
+        if (transaction != null) {
+            transaction.decrementReferences();
+        }
+        final int referenceCount = references.decrementAndGet();
+        if (referenceCount == 0) {
+            tracer.recycle(this);
```

Comment:
If end happens here as well, you won't need to manage transaction references I think as it will keep the transaction alive

Reply:
I don't quite understand

Reply:
Deferred to be considered as a future enhancement